design-doc-template.adoc

```diff
@@ -4,125 +4,95 @@ categories:
 
 === Other Interested Projects
 
 === Relevant Installation Types
-// Remove the x next to the relevant field if the feature in question is not relevant
-// to that kind of WildFly installation
-* [x] Traditional standalone server (unzipped or provisioned by Galleon)
-
-* [x] Managed domain
 
-* [x] OpenShift s2i
+__<List the installation  types thar are relevant for the features and remove any that is not relevant>__.
 
-* [x] Bootable jar
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
 
 == Requirements
 
-=== Hard Requirements
-
-=== Nice-to-Have Requirements
-// Requirements in this section do not have to be met to merge the proposed functionality.
-// Note: Nice-to-have requirements that don't end up being implemented as part of
-// the work covered by this proposal should be moved to the 'Future Work' section.
-
+__<Describe the requirements that must be fullfilled by this feature.>__
 
 === Non-Requirements
-// Use this section to explicitly discuss things that readers might think are required
-// but which are not required.
+
+__<Use this section to explicitly discuss things that readers might think are required but which are not required.>__ 
 
 === Future Work
-// Use this section to discuss requirements that are not addressed by this proposal
-// but which may be addressed in later proposals.
+
+__<Use this section to discuss requirements that are not addressed by this proposal but which may be addressed in later proposals.>__
 
 == Backwards Compatibility
 
-// Does this enhancement affect backwards compatibility with previously released
-// versions of WildFly?
-// Can the identified incompatibility be avoided?
+__<Does this enhancement affect backwards compatibility with previously released versions of WildFly? Can the identified incompatibility be avoided?>__
 
 === Default Configuration
 
+__<What is the impact of this feature in the default configuration(s) provided by WildFly?>__
+
 === Importing Existing Configuration
 
+TBD
+
 === Deployments
 
+__<What is the impact of this feature for deployments?>__
+
 === Interoperability
 
-//== Implementation Plan
-////
-Delete if not needed. The intent is if you have a complex feature which can 
-not be delivered all in one go to suggest the strategy. If your feature falls 
-into this category, please mention the Release Coordinators on the pull 
-request so they are aware.
-////
+__<Is this feature impacting interoperability?>__
 
-== Security Considerations
+== Implementation Plan
 
-////
-Identification if any security implications that may need to be considered with this feature
-or a confirmation that there are no security implications to consider.
-////
+__<This section is optional. If you have a complex feature which can not be delivered all in one go, suggest the strategy.>__
+
+== Thread Model
```

Comment:
s/Thread/Threat

I'm not sure this is a better title than "Security Considerations". I imagine many devs only have a vague idea what thread modeling is.